Publish Unit 0, Unit 1 and Bonus Unit 1 (#99)
huggingface/deep-rl-class: units/en/unit0/introduction.mdx

# Welcome to the 🤗 Deep Reinforcement Learning Course [[introduction]]

<img src="https://huggingface.co/datasets/huggingface-deep-rl-course/course-images/resolve/main/en/unit0/thumbnail.jpg" alt="Deep RL Course thumbnail" width="100%"/>

Welcome to the most fascinating topic in Artificial Intelligence: Deep Reinforcement Learning.

This course will **teach you about Deep Reinforcement Learning from beginner to expert**. It’s completely free.

In this unit you’ll:

- Learn more about the **course content**.
- **Define the path** you’re going to take (either self-audit or certification process)
- Learn more about the **AI vs. AI challenges** you're going to participate to.
- Learn more **about us**.
- **Create your Hugging Face account** (it’s free).
- **Sign-up our discord server**, the place where you can exchange with our classmates and us.

Let’s get started!

## What to expect? [[expect]]

In this course, you will:

- 📖 Study Deep Reinforcement Learning in **theory and practice.**

> - 🧑‍💻 Learn to **use famous Deep RL libraries** such as Stable Baselines3, RL Baselines3 Zoo, Sample Factory and CleanRL.

Reviewer: :heavy_check_mark:

> - 🤖 **Train agents in unique environments** such as SnowballFight, Huggy the Doggo 🐶, MineRL (Minecraft ⛏️), VizDoom (Doom) and classical ones such as Space Invaders and PyBullet.

Reviewer: :heavy_check_mark:

- 💾 Publish your **trained agents in one line of code to the Hub**. But also download powerful agents from the community.
- 🏆 Participate in challenges where you will **evaluate your agents against other teams. But also play against AI you'll train.**

And more!

At the end of this course, **you’ll get a solid foundation from the basics to the SOTA (state-of-the-art) methods**.

> You can find the syllabus on our website 👉 <a href="https://simoninithomas.github.io/deep-rl-course/">here</a>

Reviewer: No for now the website is hosting on my former Deep RL Course link

We strongly advise you to participate to the live so that you can ask questions but if you can't participate in the live, the sessions are recorded and will be posted.
- *Challenges* such AI vs. AI and leaderboard.

## Two paths: choose your own adventure [[two-paths]]

<img src="https://huggingface.co/datasets/huggingface-deep-rl-course/course-images/resolve/main/en/unit0/two-paths.jpg" alt="Two paths" width="100%"/>

You can choose to follow this course either:

- *To get a certificate of completion*: you need to complete 80% of the assignments before the end of March 2023.
- *As a simple audit*: you can participate in all challenges and do assignments if you want, but you have no deadlines.

Whatever path you choose, we advise you **to follow the recommended pace to enjoy the course and challenges with the most classmates.**
You don't need to tell us which path you choose. At the end of March, when we verify the assignments **if you get more than 80% of the assignments done, you'll get a certificate.**

## How to get most of the course? [[advice]]

To get most of the course, we have some advice:

> 1. <a href="https://discord.gg/ydHrjt3WP5">Join or create study groups in Discord </a>: studying in groups is always easier. To do that, you need to join our discord server.

Reviewer: Discord is better I'm updating it

And more!

At the end of this course, **you’ll get a solid foundation from the basics to the SOTA (state-of-the-art) methods**.

You can find the syllabus on our website 👉 <a href="https://simoninithomas.github.io/deep-rl-course/">here</a>

Don’t forget to **<a href="http://eepurl.com/ic5ZUD">sign up to the course</a>** (we are collecting your email to be able to **send you the links when each Unit is published and give you information about the challenges and updates).**

Sign up  👉 <a href="http://eepurl.com/ic5ZUD">here</a>

## What does the course look like? [[course-look-like]]
The course is composed of:

- *A theory part*: where you learn a **concept in theory (article)**.
- *A hands-on*: where you’ll learn **to use famous Deep RL libraries** to train your agents in unique environments. These hands-on will be **Google Colab notebooks but also tutorial videos**.
- *Challenges*: such AI vs. AI and leaderboard.

## Two paths: choose your own adventure [[two-paths]]

<img src="https://huggingface.co/datasets/huggingface-deep-rl-course/course-images/resolve/main/en/unit0/two-paths.jpg" alt="Two paths" width="100%"/>

You can choose to follow this course either:

> - *To get a certificate of completion*: you need to complete 80% of the assignments before the end of March 2023.

Reviewer: ✔️ Added to my list

- *As a simple audit*: you can participate in all challenges and do assignments if you want, but you have no deadlines.

Whatever path you choose, we advise you **to follow the recommended pace to enjoy the course and challenges with your fellow classmates.**
You don't need to tell us which path you choose. At the end of March, when we verify the assignments **if you get more than 80% of the assignments done, you'll get a certificate.**

## How to get most of the course? [[advice]]

To get most of the course, we have some advice:

1. <a href="https://discord.gg/ydHrjt3WP5">Join or create study groups in Discord </a>: studying in groups is always easier. To do that, you need to join our discord server.
2. **Do the quizzes and assignments**: the best way to learn is to do and test yourself.
3. **Define a schedule to stay in sync**: you can use our recommended pace schedule below or create yours.

> <img src="https://huggingface.co/datasets/huggingface-deep-rl-course/course-images/resolve/main/en/unit0/advice.jpg" alt="Course advice" width="100%"/>

Reviewer 2: Nice image :)

Reviewer: Thanks 🤗

## What tools do I need? [[tools]]

You need only 3 things:

- *A computer* with an internet connection.
- *Google Colab (free version)*: most of our hands-on will use Google Colab, the **free version is enough.**

> - A *Hugging Face Account*: to push and load models. If you don’t have an account yet you can create one here (it’s free).

Reviewer 2: ```suggestion
- A *Hugging Face Account*: to push and load models. If you don’t have an account yet, you can create one **[here](https://hf.co/join)** (it’s free).
```

<img src="https://huggingface.co/datasets/huggingface-deep-rl-course/course-images/resolve/main/en/unit0/tools.jpg" alt="Course tools needed" width="100%"/>

## What is the recommended pace? [[recommended-pace]]

We defined a planning that you can follow to keep up the pace of the course.

> <img src="https://huggingface.co/datasets/huggingface-deep-rl-course/course-images/resolve/main/en/unit0/pace1.jpg" alt="Course advice" width="100%"/>

Reviewer 2: Very nice!

> <img src="https://huggingface.co/datasets/huggingface-deep-rl-course/course-images/resolve/main/en/unit0/pace1.jpg" alt="Course advice" width="100%"/>

Reviewer 2: Maybe it's too complex to change it now, but maybe consider adding one or two gap weeks in early january.

Reviewer: ✔️ I can change the recommended pace I put it on my list

<img src="https://huggingface.co/datasets/huggingface-deep-rl-course/course-images/resolve/main/en/unit0/pace2.jpg" alt="Course advice" width="100%"/>

> Each chapter in this course is designed **to be completed in 1 week, with approximately 3-4 hours of work per week**. However, you can take as much time as you need to complete the course.

Reviewer 2: ```suggestion
Each chapter in this course is designed **to be completed in 1 week, with approximately 3-4 hours of work per week**. However, you can take as much time as necessary to complete the course. If you want to dive into a topic more in-depth, we'll provide additional resources to help you achieve that.
```

## Who are we [[who-are-we]]
About the author:

> - <a href="https://twitter.com/ThomasSimonini">Thomas Simonini</a> is a Developer Advocate at Hugging Face 🤗 specializing in Deep Reinforcement Learning. He founded Deep Reinforcement Learning Course in 2018, which became one of the most used courses in Deep RL.

Reviewer 2: ```suggestion
- <a href="https://twitter.com/ThomasSimonini">Thomas Simonini</a> is a Developer Advocate at Hugging Face 🤗 specializing in Deep Reinforcement Learning. He founded the Deep Reinforcement Learning Course in 2018, which became one of the most used courses in Deep RL.
```

> About the reviewers:

Reviewer 2: ```suggestion
About the team:
```
maybe? Not sure

> - <a href="https://twitter.com/osanseviero">Omar Sanseviero</a> is a Machine Learning engineer at Hugging Face where he works in the intersection of ML, Community and Open Source. Previously, Omar worked as a Software Engineer at Google in the teams of Assistant and TensorFlow Graphics. He is from Peru and likes llamas 🦙.

Reviewer 2: ```suggestion
- <a href="https://twitter.com/osanseviero">Omar Sanseviero</a> is a Machine Learning Engineer at Hugging Face where he works in the intersection of ML, Community and Open Source. Previously, Omar worked as a Software Engineer at Google in the teams of Assistant and TensorFlow Graphics. He is from Peru and likes llamas 🦙.
```

## When do the challenges start? [[challenges]]

In this new version of the course, you have two types of challenges:
- A leaderboard to compare your agent's performance to other classmates'.
- AI vs. AI challenges where you can train your agent and compete against other classmates' agents.

<img src="https://huggingface.co/datasets/huggingface-deep-rl-course/course-images/resolve/main/en/unit0/challenges.jpg" alt="Challenges" width="100%"/>

These AI vs.AI challenges will be announced **later in December**.

## I found a bug, or I want to improve the course [[contribute]]

Contributions are welcomed 🤗

- If you *found a bug 🐛 in a notebook*, please <a href="https://github.com/huggingface/deep-rl-class/issues">open an issue</a> and **describe the problem**.
- If you *want to improve the course*, you can <a href="https://github.com/huggingface/deep-rl-class/pulls">open a Pull Request.</a>

## I still have questions [[questions]]

> In that case, <a href="https://simoninithomas.github.io/deep-rl-course/#faq">check our FAQ</a>. And if the question is not in it, ask your question in our <a href="https://discord.gg/ydHrjt3WP5">discord server #rl-discussions.</a>

Reviewer 2: Very very very cool intro :fire:

Reviewer: Thanks 🤗
